lib/synapse/service_watcher/marathon.rb

```diff
@@ -0,0 +1,105 @@
+require "synapse/service_watcher/base"
+require 'json'
+require 'resolv'
+
+module Synapse
+  class MarathonWatcher < BaseWatcher
+    def start
+      @check_interval = @discovery['check_interval'] || 10.0
+      @connection = nil
+      @watcher = Thread.new { watch }
```

Comment:
Can we sleep for the first watch for a random amount of time between 0 and the check_interval? I feel like this could easily thunder your marathon api?